File: src/aoNewsletter2/Controllers/genericController.vb

Option Explicit On
Option Strict On

Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports Contensive.BaseClasses

Namespace Controllers
    Public Class GenericController
        '
        '====================================================================================================
        ''' <summary>
        ''' return date.minValue if date is before 1/1/1900
        ''' </summary>
        ''' <param name="sourceDate"></param>
        ''' <returns></returns>
        Public Shared Function encodeMinDate(sourceDate As Date) As Date
            If (sourceDate < #1/1/1900#) Then
                Return Date.MinValue
            End If
            Return sourceDate
        End Function

        Public Shared Function isNumeric(ByVal value As String) As Boolean
            Return value.All(AddressOf Char.IsNumber)
        End Function

        Public Shared Function isDateEmpty(ByVal srcDate As DateTime) As Boolean
            Return (srcDate < New DateTime(1900, 1, 1))
        End Function

        Public Shared Function getShortDateString(ByVal srcDate As DateTime) As String
            If Not isDateEmpty(srcDate) Then
                Return encodeMinDate(srcDate).ToShortDateString()
            End If

            Return String.Empty
        End Function

        Public Shared Function getSortOrderFromInteger(ByVal id As Integer) As String
            Return id.ToString().PadLeft(7, "0"c)
        End Function

        Public Shared Function getDateForHtmlInput(ByVal source As DateTime) As String
            If isDateEmpty(source) Then
                Return ""
            Else
                Return source.Year & "-" & source.Month.ToString().PadLeft(2, "0"c) & "-" & source.Day.ToString().PadLeft(2, "0"c)
            End If
        End Function

        Public Shared Function verifyProtocol(ByVal url As String) As String
            If (String.IsNullOrWhiteSpace(url)) Then Return String.Empty
            If (url.Substring(0, 1) = "/") Then Return url
            If (Not url.IndexOf("://").Equals(-1)) Then Return url
            Return "http://" & url
        End Function

    End Class
End Namespace

File: server/aoNewsletter2/Controllers/GenericController.cs

using System;
using System.Linq;

namespace Contensive.Addons.Newsletter.Controllers {
    public class GenericController {
        // 
        // ====================================================================================================
        /// <summary>
        /// return date.minValue if date is before 1/1/1900
        /// </summary>
        /// <param name="sourceDate"></param>
        /// <returns></returns>
        public static DateTime encodeMinDate(DateTime sourceDate) {
            if (sourceDate < DateTime.Parse("1900-01-01")) {
                return DateTime.MinValue;
            }
            return sourceDate;
        }

        public static bool isNumeric(string value) {
            return value.All(char.IsNumber);
        }

        public static bool isDateEmpty(DateTime srcDate) {
            return srcDate < new DateTime(1900, 1, 1);
        }

        public static string getShortDateString(DateTime srcDate) {
            if (!isDateEmpty(srcDate)) {
                return encodeMinDate(srcDate).ToShortDateString();
            }

            return string.Empty;
        }

        public static string getSortOrderFromInteger(int id) {
            return id.ToString().PadLeft(7, '0');
        }

        public static string getDateForHtmlInput(DateTime source) {
            if (isDateEmpty(source)) {
                return "";
            } else {
                return source.Year + "-" + source.Month.ToString().PadLeft(2, '0') + "-" + source.Day.ToString().PadLeft(2, '0');
            }
        }

        public static string verifyProtocol(string url) {
            if (string.IsNullOrWhiteSpace(url))
                return string.Empty;
            if (url.Substring(0, 1) == "/")
                return url;
            if (!url.IndexOf("://").Equals(-1))
                return url;
            return "http://" + url;
        }

    }
}